### PR TITLE
feature: npm global install support with ~/.signalk settings

### PR DESCRIPTION
**Pull request by sbender9**
The install script now creates a defaults.json with name and mmsi or uuid
This includes changes to install plugins in the config directory when it is being used.

  
Depends on #380

### CODE REVIEW

**Comment by sbender9** [comment]
Documentation still needs updated, but wanted to get feedback first...

File: bin/signalk-server-setup [hunk 1]
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+
+if [[ $EUID > 0 ]]
+  then echo "Please run as root if you want this server configuration to run at every startup, type:"
+  echo "\"sudo bash rpi-setup.sh\""
```
Comment (tkurki) [inline bin/signalk-server-setup]: Old name

**Comment by joabakk** [comment]
Good work. Rpi-installation.md still points to the old command...

**Comment by sbender9** [comment]
Currently installing globally breaks the App Store because the user does not have permission to write to node_modules.

File: lib/modules.js [hunk 2]
```diff
@@ -18,9 +18,25 @@ const fs = require('fs')
...
 const agent = require('superagent-promise')(require('superagent'), Promise)
+const path = require('path')
 
-function modulesWithKeyword (keyword) {
-  return findModulesInDir('./node_modules/', keyword)
+function modulesWithKeyword (app, keyword) {
+  var res = []
+
+  if (app.config.configPath != app.config.appPath) {
+    var pluginsPath = path.join(app.config.configPath, 'node_modules')
```
Comment (tkurki) [inline lib/modules.js]: should be `modulesPath`, since modules.js is about loading modules in general

**Pull request merged**